registry/service_v1.go

```diff
@@ -1,25 +1,13 @@
 package registry
 
-import (
-	"net/url"
-
-	"github.com/docker/go-connections/tlsconfig"
-)
+import "net/url"
 
 func (s *DefaultService) lookupV1Endpoints(hostname string) (endpoints []APIEndpoint, err error) {
-	tlsConfig := tlsconfig.ServerDefault()
-	if hostname == DefaultNamespace {
-		endpoints = append(endpoints, APIEndpoint{
-			URL:          DefaultV1Registry,
-			Version:      APIVersion1,
-			Official:     true,
-			TrimHostname: true,
-			TLSConfig:    tlsConfig,
-		})
-		return endpoints, nil
+	if hostname == DefaultNamespace || hostname == DefaultV2Registry.Host || hostname == IndexHostname {
```

Comment:
It doesn't make sense to reuse `DefaultV2Registry.Host`, since that may change for reasons unrelated to looking up v1 endpoints. Define a constant.